src/System/Hapistrano.hs

```diff
@@ -313,24 +351,25 @@ tempSymlinkPath
   -> Path Abs File
 tempSymlinkPath deployPath = deployPath </> $(mkRelFile "current_tmp")
 
--- | Get path to the directory that contains tokens of build completion.
+-- | Checks if a release was deployed properly or not
+-- by looking into the @.hapistrano_deploy_state@ file.
+-- If the file doesn't exist or the contents are anything other than
+-- 'Fail' or 'Success', it returns 'Nothing'.
 
-ctokensPath
-  :: Path Abs Dir      -- ^ Deploy path
-  -> Path Abs Dir
-ctokensPath deployPath = deployPath </> $(mkRelDir "ctokens")
-
--- | Get path to completion token file for particular release.
-
-ctokenPath
-  :: Path Abs Dir      -- ^ Deploy path
-  -> Release           -- ^ 'Release' identifier
-  -> Hapistrano (Path Abs File)
-ctokenPath deployPath release = do
-  let rendered = renderRelease release
-  case parseRelFile rendered of
-    Nothing    -> failWith 1 (Just $ "Could not append path: " ++ rendered)
-    Just rpath -> return (ctokensPath deployPath </> rpath)
+deployState
+  :: Path Abs Dir -- ^ Deploy path
+  -> Maybe (Path Rel Dir) -- ^ Working directory
+  -> Release -- ^ 'Release' identifier
+  -> Hapistrano (Maybe DeployState) -- ^ Whether the release was deployed successfully or not
+deployState deployPath mWorkingDir release = do
+  parseStatePath <- parseRelFile deployStateFilename
+  actualReleasePath <- releasePath deployPath release mWorkingDir
+  let stateFilePath = actualReleasePath </> parseStatePath
+  doesExist <- exec (CheckExists stateFilePath) (Just release)
+  if doesExist then do
+    deployStateContents <- exec (Cat stateFilePath) (Just release)
+    return $ readMaybe deployStateContents
+  else return Nothing
```

Comment:
Hey @DavidMazarro What about adding a different state for this scenario (e.g. `Unknown`):
```hs
data DeployState = Success | Failure | Unknown
```
then:
```suggestion
    fromMaybe Unknown deployStateContents
  else return Unknown
```
WDYT?